app/javascript/components/UserProfileMenu/index.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useContext } from 'react'
+import * as R from 'ramda'
+import styled from 'styled-components'
+import { withRouter } from 'react-router'
+import { useTranslation } from 'react-i18next'
+import { useMutation } from '@apollo/react-hooks'
+
+import {
+  Dropdown,
+  Trigger,
+  Menu as GardenMenu,
+} from '@zendeskgarden/react-dropdowns'
+import { Avatar } from '@zendeskgarden/react-avatars'
+import { MD } from '@zendeskgarden/react-typography'
+
+import UpdateUserOfficeMutation from '/mutations/UpdateUserOfficeMutation.gql'
+import { UserContext, FilterContext } from '/context'
+import { DefaultOfficeMenu, LanguageMenu, GeneralSettingsMenu } from './Menus'
+
+const UserDetails = styled.div`
+  margin-left: ${({ theme }) => theme.styles.spacing.sm};
+`
+
+const UserProfileContainer = styled.button`
+  display: flex;
+  border: none;
+  text-align: left;
+`
+
+const UserName = styled(MD)`
+  font-weight: bold;
+`
+
+const UserProfileMenu = ({ offices, location, router, togglePopover }) => {
+  const { currentUser, setOffice } = useContext(UserContext)
+  const { setOfficeValue } = useContext(FilterContext)
+  const [ updateDefaultOffice ] = useMutation(UpdateUserOfficeMutation)
+
+  if (R.isNil(currentUser) || R.isEmpty(currentUser)) return null
```

Comment:
did you get a warning? I don't think this fast return can go here because it creates a situation where the same number of hooks are not applied in every render. Hooks need to be applied in the same order as well.

[reference](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)

Reply:
No warning, this was just ported from the header section (which didn't display if there was no user). Would scoping it to just the `UserProfileContainer` be better?

Reply:
hmm weird. I would prefer all the hooks to be grouped up at the top though please. for consistency and readability.